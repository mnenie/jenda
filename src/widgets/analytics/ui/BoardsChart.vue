<script setup lang="ts">
import { computed, markRaw, ref } from 'vue';
import { useDark } from '@vueuse/core';
import { VueUiStackbar } from 'vue-data-ui';
import { ChartItemWrapper, type Chart } from '@/entities/chart';

const isDark = useDark();

const valuesColor = computed(() => (isDark.value ? '#a1a1aa' : '#52525b'));
const gridColor = computed(() => (isDark.value ? '#3f3f46' : '#f4f4f5'));

// @ts-ignore
const config = computed<any>(() => ({
  responsive: false,
  customPalette: [],
  useCssAnimation: true,
  orientation: 'vertical',
  table: {
    show: false
  },
  userOptions: {
    show: false
  },
  style: {
    chart: {
      backgroundColor: 'transparent',
      color: '#538BF3',
      height: 300,
      width: 800,
      padding: { top: 20, right: 36, bottom: 20, left: 36 },
      title: {
        text: '',
        color: '#1A1A1A',
        fontSize: 20,
        bold: true,
        textAlign: 'center',
        paddingLeft: 0,
        paddingRight: 0,
        subtitle: { color: '#A1A1A1', text: '', fontSize: 16, bold: false }
      },
      legend: { show: false, bold: false, backgroundColor: '#FFFFFF', color: '#1A1A1A', fontSize: 14 },
      zoom: { show: false, color: '#CCCCCC', highlightColor: '#4A4A4A', fontSize: 14, useResetSlot: false },
      tooltip: {
        show: false
      },
      highlighter: { color: '#1A1A1A', opacity: '5' },
      bars: {
        gapRatio: '0.5',
        distributed: false,
        showDistributedPercentage: true,
        borderRadius: 0,
        strokeWidth: 1,
        gradient: { show: false, intensity: '0' },
        totalValues: { show: false, offsetY: 0, fontSize: 16, bold: false, color: '#1A1A1A' },
        dataLabels: {
          show: false,
          adaptColorToBackground: true,
          color: '#1A1A1A',
          fontSize: 14,
          bold: false,
          rounding: 0,
          prefix: '',
          suffix: '',
          formatter: null
        }
      },
      grid: {
        scale: { ticks: '10' },
        x: {
          showAxis: false,
          showHorizontalLines: true,
          axisColor: gridColor.value,
          axisThickness: 2,
          axisName: { show: false, text: 'X axis', fontSize: 12, color: '#1A1A1A', bold: false, offsetY: 0 },
          timeLabels: {
            show: true,
            values: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
            offsetY: 0,
            rotation: 0,
            fontSize: 9,
            color: valuesColor.value,
            bold: false
          }
        },
        y: {
          showAxis: false,
          showVerticalLines: false,
          axisColor: '#E1E5E8',
          axisThickness: 2,
          axisName: { show: false, text: 'Y axis', fontSize: 12, color: '#1A1A1A', bold: false, offsetX: 0 },
          axisLabels: { show: true, color: valuesColor.value, fontSize: 10, bold: false }
        }
      }
    }
  }
}));

const dataset = ref([{ name: 'Series 1', series: [5, 2, 10, 2, 1, 5, 5, 9, 7, 3, 1, 4], color: '#538BF3' }]);

const chart = markRaw<Chart>({
  key: 'boards',
  section: 'workspace'
});
</script>

<template>
  <ChartItemWrapper :chart width="100%">
    <VueUiStackbar :config="config" :dataset="dataset" />
  </ChartItemWrapper>
</template>
