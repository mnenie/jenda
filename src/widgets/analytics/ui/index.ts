import UsersChart from './UsersChart.vue';

export { UsersChart };
