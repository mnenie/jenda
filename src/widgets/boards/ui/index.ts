import TeamBoards from './TeamBoards.vue';
import ActivityСharts from './ActivityСharts.vue';

export { TeamBoards, ActivityСharts };
