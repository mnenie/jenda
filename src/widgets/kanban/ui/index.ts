import KanbanWrapper from './KanbanWrapper.vue';

export { KanbanWrapper };
