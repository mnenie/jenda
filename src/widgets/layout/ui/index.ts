export * from './sidebar';
export * from './header';
