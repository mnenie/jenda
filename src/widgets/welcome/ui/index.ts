import MarketingCards from './MarketingCards.vue';
import OfferJenda from './OfferJenda.vue';
import ProjectDemo from './ProjectDemo.vue';
import MarqueeList from './MarqueeList.vue';

import KanbanSection from './sections/KanbanSection.vue';
import CollaborativeSection from './sections/CollaborativeSection.vue';
import AboutSection from './sections/AboutSection.vue';
import ActivitySection from './sections/ActivitySection.vue';
import MembersSection from './sections/MembersSection.vue';
import ChatsSection from './sections/ChatsSection.vue';

export {
  AboutSection,
  MarketingCards,
  OfferJenda,
  ProjectDemo,
  MarqueeList,
  KanbanSection,
  CollaborativeSection,
  ActivitySection,
  MembersSection,
  ChatsSection
};
