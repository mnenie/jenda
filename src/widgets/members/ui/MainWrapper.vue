<script setup lang="ts">
import { markRaw } from 'vue';
import { UiButton } from '@/shared/ui';
import { RouteNames } from '@/shared/config/consts';
import { LeaveButton } from '@/features/members/leave-team';

const links = markRaw([
  { id: '0', title: 'All participants', pathName: RouteNames.members + '.all', i18nKey: 'all.btn' },
  { id: '1', title: 'Guests', pathName: RouteNames.members + '.guests', i18nKey: 'guests.btn' }
]);
</script>
<template>
  <div :class="$style.container">
    <div :class="$style.links">
      <RouterLink v-for="link in links" :key="link.id" :to="{ name: link.pathName }" style="width: 100%">
        <UiButton
          :variant="$route.name === link.pathName ? 'outline' : 'ghost'"
          :class="{ [$style.ghost_exception]: $route.name !== link.pathName }"
        >
          <span class="text-sm">{{ $t(`members.content.${link.i18nKey}`) }}</span>
        </UiButton>
      </RouterLink>
    </div>
    <div>
      <slot />
      <div :class="$style.btns_section">
        <LeaveButton />
        <UiButton :variant="'outline'" @click="$router.push({ name: RouteNames.boards })">
          {{ $t('members.btns.view') }} (3)
        </UiButton>
      </div>
    </div>
  </div>
</template>

<style module lang="scss">
.container {
  display: flex;
  align-items: flex-start;
  gap: 30px;

  .links {
    display: flex;
    align-items: flex-start;
    gap: 10px;
    flex-direction: column;
    width: 180px;

    & button {
      width: 100%;
      justify-content: flex-start;
    }
    .ghost_exception {
      border: 1px solid transparent !important;
    }
  }

  .btns_section {
    display: flex;
    align-items: center;
    gap: 20px;
    margin-top: 50px;
  }
}
</style>
