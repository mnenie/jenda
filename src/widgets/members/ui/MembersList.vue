<script setup lang="ts">
import { UserAvatar } from '@/entities/user';
import type { UserType } from '@/entities/user/model';
import { UiBadge } from '@/shared/ui';
import { useI18n } from 'vue-i18n';

defineProps<{
  members: UserType[];
}>();

const { t } = useI18n();
</script>

<template>
  <ul :class="$style.list">
    <li v-for="member in members" :key="member._id">
      <div :class="$style.info">
        <UserAvatar>
          <img :src="member.photoUrl" alt="member" loading="lazy" />
        </UserAvatar>
        <p class="text-sm">{{ member.email }}</p>
      </div>
      <UiBadge :variant="'secondary'">{{ t(`members.role.${member.role}`) }}</UiBadge>
    </li>
  </ul>
</template>

<style module lang="scss">
.list {
  display: flex;
  flex-direction: column;
  gap: 15px;
  width: 400px;

  .info {
    display: flex;
    gap: 8px;
    align-items: center;
  }

  & li {
    display: flex;
    align-items: center;
    justify-content: space-between;
    gap: 4px;
  }

  & img {
    width: 100%;
  }
}
</style>
