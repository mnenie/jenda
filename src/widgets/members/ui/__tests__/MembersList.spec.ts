import { shallowMount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import MembersList from '../MembersList.vue';
import { UiBadge } from '@/shared/ui';
import { UserAvatar } from '@/entities/user';

describe('tests for MembersList.vue', () => {
  const wrapper = shallowMount(MembersList, {
    props: {
      members: [
        {
          _id: '1',
          email: 'test@gmail.com'
        }
      ]
    }
  });

  it('it should be render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render subcomponents', () => {
    expect(wrapper.findComponent(UiBadge).exists()).toBe(true);
    expect(wrapper.findComponent(UserAvatar).exists()).toBe(true);
  });
});
