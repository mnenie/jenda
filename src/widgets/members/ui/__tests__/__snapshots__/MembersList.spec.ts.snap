// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for MembersList.vue > it should be render correctly 1`] = `
"<ul class="list">
  <li>
    <div class="info">
      <anonymous-stub></anonymous-stub>
      <p class="text-sm">test@gmail.com</p>
    </div>
    <ui-badge-stub variant="secondary"></ui-badge-stub>
  </li>
</ul>"
`;
